src/main/java/com/ldtteam/structurize/storage/StructurePackMeta.java

```diff
@@ -0,0 +1,213 @@
+        final JsonArray authorArray = json.get("authors").getAsJsonArray();
+        for (int i = 0; i < authorArray.size(); i++)
+        {
+            authors.add(authorArray.get(i).getAsString());
+        }
+
+        this.desc = json.get("desc").getAsString();
+
+        this.mods = new ArrayList<>();
+        final JsonArray modsArray = json.get("mods").getAsJsonArray();
+        for (int i = 0; i < modsArray.size(); i++)
+        {
+            mods.add(modsArray.get(i).getAsString());
+        }
+        this.rootPath = rootPath;
+        this.version = json.get("version").getAsInt();
+        this.packFormat  = json.get("pack-format").getAsInt();
+    }
+
+    public String getName()
+    {
+        return this.packName;
+    }
+
+    /**
+     * Set the immutability param.
+     *
+     * @param immutable true if immutable, else false.
+     */
+    public void setImmutable(final boolean immutable)
+    {
+        this.immutable = immutable;
+    }
+
+    /**
+     * Check if the structure pack is immutable.
+     *
+     * @return true if so.
+     */
+    public boolean isImmutable()
+    {
+        return this.immutable;
+    }
+
+    /**
+     * Get the pack format.
+     *
+     * @return the format version.
+     */
+    public int getPackFormat()
+    {
+        return this.packFormat;
+    }
+
+    /**
+     * Get the pack version.
+     *
+     * @return the pack version.
+     */
+    public int getVersion()
+    {
+        return this.version;
+    }
+
+    /**
+     * Get the list of mods that were used in this schematic.
+     * @return the list of mod ids.
+     */
+    public List<String> getModList()
+    {
+        return mods;
+    }
+
+    /**
+     * Get the structure pack root path.
+     */
+    public Path getPath()
+    {
+        return this.rootPath;
+    }
+
+    /**
+     * Get the icon path string.
+     * @return the name of the icon.
+     */
+    public String getIconPath()
+    {
+        return this.iconPath;
+    }
+
+    /**
+     * Getter for the pack description.
+     * @return the desc string.
+     */
+    public String getDesc()
+    {
+        return this.desc;
+    }
+
+    /**
+     * Get the list of authors.
+     * @return list of names.
+     */
+    public List<String> getAuthors()
+    {
+        return this.authors;
+    }
+
+    /**
+     * Get the sub path from this pack.
+     * @param subPath the path we're in.
+     * @return the relative sub path.
+     */
+    public String getSubPath(final Path subPath)
+    {
+        if (rootPath.toString().contains("\\"))
+        {
+            return subPath.toString().replace(rootPath + "\\", "");
+        }
+        else
+        {
+            return subPath.toString().replace(rootPath + "/", "");
+        }
+    }
+
+    /**
+     * Get the normalized sub path from this pack.
+     * @param subPath the path we're in.
+     * @return the relative sub path.
+     */
+    public String getNormalizedSubPath(final String subPath)
```

Comment:
It's for windows/linux server/client sync of subpaths

Reply:
like if its blueprint coordinate then it should have fixed logic like ResLoc do, if not then I would expect nio Path being used, this really smells fishy

Reply:
Yeah, it's a pain, we sync it from client to server (and from server to client) so we can't use a path so we can't use NIO values but we can't really use res locations either because we might use invalid chars in it (afaik at least). And on the top of that in jar locations on windows are "/" and out of the jar in "\"

Reply:
well I know syncing paths suck a lot, but that's why we should do stable system for it, and honestly meh invalid chars, I would go with same (or maybe a bit looser, but no acutes etc. and always lowercase) restrictions as in resLoc

Reply:
but what if there are folders that don't respect that. So players would need to respect that with folder and file names too, which is a pita. I know that this is not perfect. But I never intended it to be perfect nor "100% finished". If I did 100% finished, I'd be done next year. But it doesn't have to be 100% finished to be better than the status quo, flexible enough to allow decent improvements and feature complete